src/pages/index.js

```diff
@@ -61,16 +107,64 @@ const addImageModal = new ModalWithForm(
 //-----------CREATE A MODALWITHIMAGE INSTANCE FOR THE PREVIEW IMAGE------------>>
 const previewModal = new ModalWithImage("#image-preview-modal");
 
+//-----------CREATE A FUNCTION THAT HANDLES DELETE CLICK------------>>
+function handleDeleteClick(card) {
+  deleteConfirmationModal.open();
+  deleteConfirmationModal.setCallback(() => {
+    api.deleteCard(card.getId()).catch((err) => console.log(err));
+    card.deleteCard();
+    deleteConfirmationModal.close();
+  });
+}
+
+//-----------CREATE A FUNCTION THAT HANDLES LIKE CLICK------------>>
+function handleLikeClick(card) {
+  api
+    .likeCard(card.getId(), card.isLiked)
+    .then((res) => card.toggleLikeCard(res.isLiked))
+    .catch((err) => console.log(err));
+}
+
+//-----------CREATE A FUNCTION THAT HANDLES AVATAR EDIT SUBMIT------------>>
+function handleAvatarFormSubmit(values) {
+  avatarEditModal.renderSaving(true);
+  api
+    .changeAvatar(values)
+    .then((res) => {
+      userInfo.setUserAvatar(res.avatar);
+    })
+    .catch((err) => console.log(err))
+    .finally(avatarEditModal.renderSaving(false));
+  avatarEditModal.close();
+}
+
 //-----------CREATE A FUNCTION THAT HANDLES PROFILE EDIT SUBMIT------------>>
 function handleProfileFormSubmit(values) {
-  userInfo.setUserInfo(values);
+  profileEditModal.renderSaving(true);
+  api
+    .editProfile(values)
+    .then((res) => {
+      userInfo.setUserInfo({
+        name: res.name,
+        description: res.about,
+      });
+    })
+    .catch((err) => console.log(err))
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors